### PR TITLE
Add set settings from reader

### PR DESCRIPTION
**Pull request by kropptrevor**

### CODE REVIEW

File: repositories/settings/core.go [hunk 1]
```diff
@@ -22,3 +37,46 @@ func (c *core) setResolution(width, height int32) {
...
+
+func (c *core) setFromReader(reader io.Reader) error {
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+		err := scanner.Err()
+		if err != nil {
+			return err
+		}
+		if strings.Trim(line, " \t") == "" {
```
Comment (kroppt) [inline repositories/settings/core.go:49]: https://pkg.go.dev/strings@go1.17#TrimSpace

File: repositories/settings/core.go [hunk 2]
```diff
@@ -22,3 +37,46 @@ func (c *core) setResolution(width, height int32) {
...
+			continue
+		}
+		elements := strings.Split(line, "=")
+		if len(elements) != 2 {
+			return errors.New("malformed settings line: expected key=value")
```
Comment (kroppt) [inline repositories/settings/core.go:54]: This should be a sentinel error.
This error is not tested.

**Review by kroppt — CHANGES_REQUESTED** [review]

**Review by gregjohnson2017 — DISMISSED** [review]

File: repositories/settings/core.go (hunk 1 above)
Comment (kropptrevor) [reply repositories/settings/core.go]: This has been updated.

File: repositories/settings/core.go (hunk 2 above)
Comment (kropptrevor) [reply repositories/settings/core.go]: This has been updated.

Comment (gregjohnson2017) [reply repositories/settings/core.go]: I don't think we should be using sentinel errors. I was convinced by this article: https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully

**Comment by kropptrevor** [comment]
We need an in depth discussion on error handling, in particular how it relates to testing and coupling.

**Review by kroppt — APPROVED** [review]

**Review by gregjohnson2017 — APPROVED** [review]

**Pull request merged**